packages/strapi-plugin-graphql/services/Resolvers.js

```diff
@@ -404,45 +406,44 @@ module.exports = {
                       return related[ref.primaryKey] || related;
                     }
                   );
+                  
+                  Object.assign(queryOpts, {
+                    ...queryOpts,
+                    query: {
+                      [ref.primaryKey]: arrayOfIds
+                    }
+                  });
 
-                  // Where.
-                  queryOpts.query = strapi.utils.models.convertParams(name, {
-                    // Construct the "where" query to only retrieve entries which are
-                    // related to this entry.
-                    [ref.primaryKey]: arrayOfIds,
-                    ...where.where,
-                  }).where;
                   break;
                 }
                 default:
-                  // Where.
-                  queryOpts.query = strapi.utils.models.convertParams(name, {
-                    // Construct the "where" query to only retrieve entries which are
-                    // related to this entry.
-                    [association.via]: obj[ref.primaryKey],
-                    ...where.where,
-                  }).where;
+                  Object.assign(queryOpts, {
+                    ...queryOpts,
+                    query: {
+                      [association.via]: obj[ref.primaryKey]
+                    }
+                  });
               }
             }
 
             if (queryOpts.hasOwnProperty('query') &&
               queryOpts.query.hasOwnProperty('id') &&
               queryOpts.query.id.hasOwnProperty('value') &&
               Array.isArray(queryOpts.query.id.value)
-            ){
+            ) {
               queryOpts.query.id.symbol = 'IN';
             }
 
-            const value = await (association.model
-              ? resolvers.fetch(params, association.plugin, [])
-              : resolvers.fetchAll(params, { ...queryOpts, populate: [] }));
+            const loaderName = association.plugin ? `${association.plugin}__${params.model}`: params.model;
```

Comment:
This needs to go in a utility function, because you're duplicating a logic twice in two different places (It'll likely break some day)